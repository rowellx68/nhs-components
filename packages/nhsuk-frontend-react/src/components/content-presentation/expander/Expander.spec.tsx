import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, {
  SingleExpander as SingleExpanderStory,
  MoreThanOneExpander as MoreThanOneExpanderStory,
} from './Expander.stories';

const SingleExpander = composeStory(SingleExpanderStory, meta);
const MoreThanOneExpander = composeStory(MoreThanOneExpanderStory, meta);

it('should render the Expander component', () => {
  const { container } = render(<SingleExpander />);

  expect(container.querySelector('.nhsuk-expander')).toBeInTheDocument();
  expect(container.querySelector('summary')).toHaveTextContent(
    'Get your medical records',
  );
  expect(container).toMatchSnapshot();
});

it('should render the Expander Group component', () => {
  const { container } = render(<MoreThanOneExpander />);

  expect(container.querySelector('.nhsuk-expander-group')).toBeInTheDocument();
  expect(container.querySelectorAll('.nhsuk-expander')).toHaveLength(2);
  expect(container).toMatchSnapshot();
});
