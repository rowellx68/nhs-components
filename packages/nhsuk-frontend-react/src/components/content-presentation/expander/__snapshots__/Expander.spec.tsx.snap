// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the Expander Group component 1`] = `
<div>
  <div
    class="nhsuk-expander-group"
  >
    <details
      class="nhsuk-details nhsuk-expander"
      id="nhsuk-details0"
      nhsuk-polyfilled="true"
    >
      <summary
        aria-controls="nhsuk-details__text0"
        aria-expanded="false"
        class="nhsuk-details__summary"
        role="button"
        tabindex="0"
      >
        <span
          class="nhsuk-details__summary-text"
        >
          How to measure your blood glucose levels
        </span>
      </summary>
      <div
        aria-hidden="true"
        class="nhsuk-details__text"
        id="nhsuk-details__text0"
        style="display: none;"
      >
        <p>
          Testing your blood at home is quick and easy, although it can be uncomfortable. It does get better.
        </p>
        <p>
          You would have been given:
        </p>
        <ul>
          <li>
            a blood glucose metre
          </li>
          <li>
            small needles called lancets
          </li>
          <li>
            a plastic pen to hold the lancets
          </li>
          <li>
            small test strips
          </li>
        </ul>
      </div>
    </details>
    <details
      class="nhsuk-details nhsuk-expander"
      id="nhsuk-details1"
      nhsuk-polyfilled="true"
    >
      <summary
        aria-controls="nhsuk-details__text1"
        aria-expanded="false"
        class="nhsuk-details__summary"
        role="button"
        tabindex="0"
      >
        <span
          class="nhsuk-details__summary-text"
        >
          When to check your blood glucose level
        </span>
      </summary>
      <div
        aria-hidden="true"
        class="nhsuk-details__text"
        id="nhsuk-details__text1"
        style="display: none;"
      >
        <p>
          Try to check your blood:
        </p>
        <ul>
          <li>
            before meals
          </li>
          <li>
            2 to 3 hours after meals
          </li>
          <li>
            before, during (take a break) and after exercise
          </li>
        </ul>
        <p>
          This helps you understand your blood glucose levels and how theyâ€™re affected by meals and exercise. It should help you have more stable blood glucose levels.
        </p>
      </div>
    </details>
  </div>
</div>
`;

exports[`should render the Expander component 1`] = `
<div>
  <details
    class="nhsuk-details nhsuk-expander"
    id="nhsuk-details0"
    nhsuk-polyfilled="true"
  >
    <summary
      aria-controls="nhsuk-details__text0"
      aria-expanded="false"
      class="nhsuk-details__summary"
      role="button"
      tabindex="0"
    >
      <span
        class="nhsuk-details__summary-text"
      >
        Get your medical records
      </span>
    </summary>
    <div
      aria-hidden="true"
      class="nhsuk-details__text"
      id="nhsuk-details__text0"
      style="display: none;"
    >
      <p>
        You can see your GP records by:
      </p>
      <ul>
        <li>
          asking for them at your GP surgery 
        </li>
        <li>
          going online to see them (if you have signed up for
           
          <a
            class="nhsuk-link"
            href="/using-the-nhs/nhs-services/gps/gp-online-services/"
          >
            GP online services
          </a>
          )
        </li>
      </ul>
    </div>
  </details>
</div>
`;
